firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/CheckForUpdateClient.java

```diff
@@ -135,11 +147,27 @@ private boolean isNewerBuildVersion(AppDistributionReleaseInternal latestRelease
         > getInstalledAppVersionCode(firebaseApp.getApplicationContext());
   }
 
-  private boolean isInstalledRelease(AppDistributionReleaseInternal latestRelease) {
+  @VisibleForTesting
+  boolean isInstalledRelease(AppDistributionReleaseInternal latestRelease) {
     if (latestRelease.getBinaryType().equals(BinaryType.APK)) {
-      // TODO(rachelprince): APK codehash verification. For now assume
-      // the release is identical unless the build version is different
-      return true;
+      Context context = firebaseApp.getApplicationContext();
```

Comment:
This block of code is getting pretty long. Maybe we should extract a method for lines 153-170